documentation/SupportingDocuments/Purpose Consent Proposal comparison.md

```diff
@@ -0,0 +1,34 @@
+| Do link between OPGW product and purpose is visible in Telco Catalog? |  |  In the catalog, it could be possible to browse OPGW product per category. Several categories could be created, as such 'purpose' could be defined as a category, and purpose  value could be defined as category value (in order to get for example OPGW Product related to Fraud prevention & detection) but this is only informative information.  |
+| What are country market required agreement before to launch?|   | Telco must agree on the OPGW product to be launched in the country as a minimum 'critical mass' must be available. Additionally Telco must indicate for each API scope the legal base associated with purposes for this country.|
+| What is expected during Application onboarding?|   | When a developer wishes to uses CAMARA API, developer first onboards an application. Mandatory information to onboard an application is to provide the purpose (or the list of purposes) associated with this application. |
+| What is expected for each `/authorize` call? |   | For each `/authorize` call the `scope` attribute **must** be valued. in Orange proposal, this scope is valued by a concatenation of purpose+technical scope. This is straightforward for the developer to build it as: <br> - the purpose is known from the app onboarding (see previous point)  <br> - the technical scope is  defined in each CAMARA API yaml in the `security` property at path/method level. <br> Note: For one `/authorize` it is valid to ask for several technical scopes like for example `scope=FraudDetectionAndPrevention:simswap-verify-read  number-verification-verify-read`. This is a recommended pattern to limit interaction with end user. |
+| How the legal consent displayed to the end user is defined and provided? | | Telco  associates a legal wording for each scope design-time. Then on the fly the legal notice is generated and displayed to the front end. <br> <br > As an example, simswap-verify-read technical scope is associated to wording:  'Verify latest SIM swap Date', number-verification-verify-read to 'Verify your MSIDN from network'. The request `POST /authorize` with `scope=FraudDetectionAndPrevention:simswap-verify-read` triggers following notice: <br> _"For fraud detection and prevention, application NiceApp request your consent to perform following action(s): <br> - Verify latest SIM swap Date <br> - Verify your MSIDN from network"_

```

Comment:
With this solution, would it be also allowed to send within the same post request, several different purposes? For eg. FraudDetectionAndPrevention and ServiceOptimization?

Reply:
@shilpa-padgaonkar No but I'm not sure about the UC you have in mind. When an `authorize` is send for claiming data access to end user data from an app I'm not sure how we can have several purpose. I understood we can have an app with several purpose but for one given call only one purpose should be passed. I update the table accordingly.

Reply:
I am glad to hear this! We can ONLY support the scenario if exactly one purpose is sent for a given call.